File: packages/firestore/ios/Plugin/Classes/Constraints/QueryOrderByConstraint.swift
import Foundation
import FirebaseFirestore
import Capacitor

@objc public class QueryOrderByConstraint: NSObject, QueryNonFilterConstraint {
    private var fieldPath: String
    private var directionStr: String

    public init(_ queryConstraint: JSObject) {
        self.fieldPath = queryConstraint["fieldPath"] as? String ?? ""
        self.directionStr = queryConstraint["directionStr"] as? String ?? ""
    }

    public func toQuery(query: Query) async throws -> Query {
        return query.order(by: self.fieldPath, descending: self.directionStr == "desc" ? true : false)
    }
}

File: packages/firestore/android/src/main/java/app/capgo/capacitor/firebase/firestore/QueryOrderByConstraint.java
package app.capgo.capacitor.firebase.firestore.classes.constraints;

import app.capgo.capacitor.firebase.firestore.interfaces.QueryNonFilterConstraint;
import com.getcapacitor.JSObject;
import com.google.firebase.firestore.Query;

public class QueryOrderByConstraint implements QueryNonFilterConstraint {

    private String fieldPath;
    private String directionStr;

    public QueryOrderByConstraint(JSObject queryConstraint) {
        this.fieldPath = queryConstraint.getString("fieldPath");
        this.directionStr = queryConstraint.getString("directionStr");
    }

    public String getFieldPath() {
        return fieldPath;
    }

    public String getDirectionStr() {
        return directionStr;
    }

    public Query toQuery(Query query, com.google.firebase.firestore.FirebaseFirestore firestoreInstance) {
        Query.Direction direction = directionStr.equals("desc") ? Query.Direction.DESCENDING : Query.Direction.ASCENDING;
        return query.orderBy(fieldPath, direction);
    }
}
